File: server/AppSupport.Data/Entities/Person.cs
using System;
using System.Collections.Generic;

namespace AppSupport.Data.Entities
{
    public class Person
    {
        public int Id { get; set; }
        public int? ExecutiveId { get; set; }
        public int OrganizationId { get; set; }
        public int RankId { get; set; }
        public int DodId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string MiddleName { get; set; }
        public string Nickname { get; set; }
        public string Occupation { get; set; }
        public string Ssn { get; set; }
        public string Title { get; set; }

        public Person Executive { get; set; }
        public Organization Organization { get; set; }
        public Rank Rank { get; set; }

        public IEnumerable<Person> Associates { get; set; }
        public IEnumerable<ManifestPerson> ManifestPlanes { get; set; }
        public IEnumerable<TemplatePerson> TemplatePlanes { get; set; }
        public IEnumerable<ManifestPerson> Trips { get; set; }
    }
}

File: client/core/models/api/person.ts
import {
  FormBuilder,
  FormGroup,
  Validators
} from '@angular/forms';

import { ManifestPerson } from './manifest-person';
import { Organization } from './organization';
import { Rank } from './rank';
import { TemplatePerson } from './template-person';

export interface Person {
  id: number;
  executiveId?: number;
  organizationId: number;
  rankId: number;
  dodId: number;
  firstName: string;
  lastName: string;
  middleName: string;
  nickname: string;
  occupation: string;
  ssn: string;
  title: string;

  executive: Person;
  organization: Organization;
  rank: Rank;

  associates: Person[];
  personManifestPlanes: ManifestPerson[];
  personTemplatePlanes: TemplatePerson[];
  trips: ManifestPerson[];
}

export const PersonForm = (person: Person, fb: FormBuilder, ssnPattern: RegExp): FormGroup =>
  fb.group({
    id: [person.id],
    executiveId: [person.executiveId],
    organizationId: [person.organizationId, Validators.required],
    rankId: [person.rankId, Validators.required],
    dodId: [person.dodId, [
      Validators.required,
      Validators.pattern(/^\d{10}$/g)
    ]],
    firstName: [person.firstName, Validators.required],
    lastName: [person.lastName, Validators.required],
    middleName: [person.middleName],
    nickname: [person.nickname, Validators.required],
    occupation: [person.occupation, Validators.required],
    ssn: [person.ssn, [
      Validators.required,
      Validators.pattern(ssnPattern)
    ]],
    title: [person.title, Validators.required]
  });
